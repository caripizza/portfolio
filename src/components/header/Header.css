.Header {
  z-index: 1;

  a {
    border-bottom-left-radius: 50%;
    border-top-right-radius: 50%;
    padding: .5em;
  }
  a, button {
    color: var(--black-100);
    background: var(--white-100);
    -webkit-transition-property: background;
    -moz-transition-property: background;
    -o-transition-property: background;
    transition-property: background;
    -webkit-transition-duration: 1s;
    -moz-transition-duration: 1s;
    -o-transition-duration: 1s;
    transition-duration: 1s;
    -webkit-transition-timing-function: linear;
    -moz-transition-timing-function: linear;
    -o-transition-timing-function: linear;
    transition-timing-function: linear;
  }
  & a:hover,
  & button[data-style="toggle-dark-mode"]:hover {
    background-color: var(--red-500);
  }

  ul {
    width: 50%;
    float: right;
    list-style: none;
    display: flex;
    justify-content: space-around;
    align-items: center;
    padding: 2em 0 1em 0;
    margin-top: .2em;
    height: 1em;
  }

  h1 {
    font-weight: 300;
    font-size: 1.4em;
    letter-spacing: .1em;
    width: 49%;
    float: left;
    padding: .34em 0 0 1em;
    
    a {
      background: transparent;
      text-decoration: none;
    }
  }

  h1:hover {
    color: var(--red-500);
  }

  @media only screen and (max-width: 330px) and (min-width: 200px) {
    h1 {
      font-size: 1.2em;
      padding: .6em 0 0 1em;
    }
  }

  @media screen and (max-width: 600px) {
    h1 {
      width: auto;
      padding-bottom: .3em;
    }
  }
}
