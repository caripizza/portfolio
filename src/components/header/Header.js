import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Header.css';
import SideBar from '../sidebar/SideBar';

export default class Header extends React.Component {
  state = {
    width: window.innerWidth
  };

  UNSAFE_componentWillMount() {
    window.addEventListener('resize', this.handleWindowSizeChange);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleWindowSizeChange);
  }

  handleWindowSizeChange = () => {
    this.setState({ width: window.innerWidth });
  };

  render() {
    const { width } = this.state;
    const isMobile = width <= 600;

    return (
      <>
        <header className={styles.Header}>
          <nav>
            {isMobile ? <SideBar pageWrapId={'page-wrap'} /> : null}
            <h1>
              <NavLink to="/">CARI PALAZZOLO</NavLink>
            </h1>

            {isMobile ? null : (
              <ul>
                <li>
                  <NavLink to="/">Home</NavLink>
                </li>
                <li>
                  <NavLink to="/projects">Projects</NavLink>
                </li>
                <li>
                  <a
                    href="https://cari.pizza"
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    Art/Music
                  </a>
                </li>
              </ul>
            )}
          </nav>
        </header>
      </>
    );
  }
}
