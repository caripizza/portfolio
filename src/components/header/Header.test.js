import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  it('matches a snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
