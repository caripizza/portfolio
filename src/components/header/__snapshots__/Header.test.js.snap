// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches a snapshot 1`] = `
<header
  className="Header"
>
  <nav>
    <h1>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        CARI PALAZZOLO
      </a>
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
      </li>
      <li>
        <a
          href="https://cari.pizza"
          rel="noopener noreferrer"
          target="_blank"
        >
          Art/Music
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
